activation/handler.go

```diff
@@ -82,7 +109,48 @@ func NewHandler(
 		inProgress: make(map[types.ATXID][]chan error),
 	}
 
-	return h
+	for _, opt := range opts {
+		opt(h)
+	}
+	if err := verifyAtxVersions(h.versions); err != nil {
+		return nil, fmt.Errorf("invalid ATX versions configured (%v): %w", h.versions, err)
+	}
+
+	h.log.With().Info("atx handler created",
+		log.Array("supported ATX versions", log.ArrayMarshalerFunc(func(enc zapcore.ArrayEncoder) error {
+			for _, v := range h.versions {
+				enc.AppendString(fmt.Sprintf("v%v from epoch %d", v.AtxVersion, v.publish))
+			}
+			return nil
+		})))
+
+	return h, nil
+}
+
+func verifyAtxVersions(versions []atxVersion) error {
+	if len(versions) == 0 {
+		return errors.New("no ATX versions configured")
+	}
+	slices.SortFunc(versions, func(a, b atxVersion) int {
+		switch {
+		case a.publish < b.publish:
+			return -1
+		case a.publish > b.publish:
+			return 1
+		}
+		return 0
+	})
```

Comment:
NIT, can be simplified to:

```suggestion
	slices.SortFunc(versions, func(a, b atxVersion) int { return a.publish - b.publish })
```

Reply:
That won't work because `types.EpochID` is an unsigned integer. It could underflow and return a "big" number instead of a negative. It also needs to be converted to int64/int first.